pkgs/applications/networking/instant-messengers/teams/default.nix

```
@@ -14,109 +14,150 @@
 , xdg-utils
 , systemd
 , nodePackages
-, enableRectOverlay ? false }:
+, xar
+, cpio
+, makeWrapper
+, enableRectOverlay ? false
+}:
 
-stdenv.mkDerivation rec {
+let
   pname = "teams";
   version = "1.4.00.26453";
-
-  src = fetchurl {
-    url = "https://packages.microsoft.com/repos/ms-teams/pool/main/t/teams/teams_${version}_amd64.deb";
-    sha256 = "0ndqk893l17m42hf5fiiv6mka0v7v8r54kblvb67jsxajdvva5gf";
+  meta = with lib; {
+    description = "Microsoft Teams";
+    homepage = "https://teams.microsoft.com";
+    downloadPage = "https://teams.microsoft.com/downloads";
+    license = licenses.unfree;
+    maintainers = [ maintainers.liff ];
+    platforms = platforms.linux ++ platforms.darwin;
```

`platforms.linux` is too broad since the package is only available for x86_64.

Also, `platforms.darwin` contains:

```nix
[ "x86_64-darwin" "i686-darwin" "aarch64-darwin" "armv7a-darwin" ]
```

so perhaps the list should be:

```suggestion
    platforms = [ "x86_64-linux" "x86_64-darwin" "aarch64-darwin" ];
```